src/context.js

```diff
@@ -1,3 +1,21 @@
-import React from 'react';
+import React, { useContext } from 'react';
 
-export const { Consumer, Provider } = React.createContext([]);
+const FeaturesContext = React.createContext([]);
+
+export const { Consumer, Provider } = FeaturesContext;
+
+export function useFeatures() {
+  return useContext(FeaturesContext);
+}
+
+export function withFeatures(Component) {
+  return props => {
+    return (
+      <FeaturesContext.Consumer>
+        {features => (
+          <Component {...props} features={features ? features : {}} />
```

Comment:
I don't think we need a default for features here because context defaults to an empty array.

```suggestion
          <Component {...props} features={features} />
```